scripts/lib/CIME/XML/env_batch.py

```diff
@@ -740,15 +756,11 @@ def compare_xml(self, other):
                                               root=node, otherroot=f2group))
         return xmldiffs
 
-    def make_all_batch_files(self, case, test_mode=False):
-        testcase = case.get_value("TESTCASE")
+    def make_all_batch_files(self, case):
+        is_test = case.get_value("TEST")
         machdir  = case.get_value("MACHDIR")
-        logger.info("Creating batch script case.run")
+        logger.info("Creating batch scripts")
         for job in self.get_jobs():
             input_batch_script = os.path.join(machdir,self.get_value('template', subgroup=job))
-            if job == "case.test" and testcase is not None and not test_mode:
```

Comment:
Why were we not regenerating case.test in test_mode?

Reply:
Because in some cases, and these may all be gone now that we do two case tests, we needed to change the run script mid test but keep the test script the same.